src/checkers/inference/model/ConstraintManager.java

```diff
@@ -167,6 +165,17 @@ private AnnotationLocation getCurrentLocation() {
         }
     }
 
+    // visitorState.getPath().getLeaf() caused NullPointerException on some real project
+    // and doesn't always return correct source tree. Using class tree at least shows which
```

Comment:
1. "Using class tree at least shows " is ambiguous -- i.e. the code in this function actually only use class tree when `visitorState.getPath()` is `null`.

Maybe only add a code comment on the else branch to document using class tree is a workaround for when `visitorState.getPath()` is `null`?

2. " and doesn't always return correct source tree": do you have minimum code example for that? Should this be an issue instead of a code comment?